stats/registry.go

```diff
@@ -0,0 +1,133 @@
+package stats
+
+import (
+	"expvar"
+	"sync"
+)
+
+// A singleton reference to the stats registry
+var Root Registry = &registry{
+	listeners: []func(Openable){},
+}
+
+// Provide an interface
+type Registry interface {
+
+	// Called to obtain a builder of new Statistics object.
+	NewBuilder(k string, n string, tags map[string]string) Builder
+
+	// Notify the registry that statistics object has been opened for the first time.
+	NotifyOpen(s Statistics)
+
+	// Register a listener for NotifyOpen events.
+	OnOpen(listener func(s Openable))
+
+	// Called to iterate over the registered statistics sets
+	Do(f func(s Statistics))
+}
+
+// manages the registry of all statistics sets
+type registry struct {
+	mu        sync.RWMutex
+	listeners []func(Openable)
+}
+
+// ensure the top level map is always registered
+func init() {
+	m := &expvar.Map{} // this map can't be replaced because it is a top level map
+	m.Init()
+
+	r := &expvar.Map{} // this map is replaceable, since it is a value of a map
+	r.Init()
+
+	m.Set("root", r)
+	expvar.Publish("influx", m)
+}
+
+// iterate over the registry executing the function
+// specified. after it executes, we check to see if
+// the set is now closed. if it is, then we
+// we rebuild the top level map.
+func (r *registry) Do(f func(s Statistics)) {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+
+	closed := map[string]bool{}
+	r.getRoot().Do(func(kv expvar.KeyValue) {
+		set := kv.Value.(Statistics)
+		if set.Refs() > 0 {
+			f(set)
+		}
+		if set.Refs() == 0 {
+			closed[kv.Key] = true
+		}
+	})
+
+	// rebuild the registry map
+	if len(closed) > 0 {
```

Comment:
make this literal a named constant, and move the documentation of what it is for there.